doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+   
+2. As a client I want to `Get` sensor values or actuator current values of the vehicle and its environment in a uniform manner.
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the paths of the requested signals, which optionally may contain wildcards.
+    * I want to receive exactly all the signals’ values I requested without errors.
+  
+3. As a client I want to `Subscribe` to sensor values or actuator current values when some signal value changed -> what about continuous subscription?
+  - Details:
+    * All sensor and actuator values for which a client have permission to.
+    * Client must provide the path, paths or wildcard paths.
+    * I want to receive exactly all the signals’ values I requested without errors.
+
+4. As a client I want to `Subscribe` and be able to apply a filter to receive a sensor value with an interval of x milliseconds only if the value has changed.
+   
+5.  As a client I want to `GetMetadata` static Metadata from signals.
+  - Details:
+    * All sensors, actuators, and attributes values for which a client have permission to.
+    * Client must provide the path, paths or wildcard.
+
+6.  As a client I want to `Get` attributes values once.
+
+7.  As a client I want to `Actuate` a target value of an actuator and guarantee that the operation succeeded.
+
+8.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+10.  As a client I want to know if there is a provider up and running by getting some sensor or actuator value.
+
+11.  As a client I don’t want to `Subscribe` to target values.
+
+12.  As a client I don’t want to `Subscribe` to attributes values.
+
+13.  As a client I don’t want to `Get` target values.
+
+14.  As a client I don’t want to `Set` sensor values.
+
+15.  As a client I don’t want to `Set` attributes values.
+
+### As Databroker
+1. The data broker shall be able to handle catalogs of signals described by the syntax as defined by the COVESA Vehicle Signal Specification (VSS). This relates to all aspects of the VSS syntax definition, which is also called VSS rule set. This implies that the data broker can handle the signal catalog as defined by the COVESA VSS.
+
+2. Databroker shall support at least those metadata elements as defined by the VSS rule set.
+
+### As Provider
+1. As a provider I want to `Set` the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+2. 
+3. As a provider I want to `Update` using stream atomically the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+
+4. As a provider I want to `Subscribe` to actuator target values requests.
+
+5. As a provider I want to `Set` once attributes values.
+
+6. As a provider I don’t need to but I would like to `Get` sensor values or current values of actuatros?.
```

Comment:
I see _consumer_ and _provider_ as separate roles - but not as separate components/applications: A component can have one of those roles or both, i.e. being pure consumer or pure provider, or a combined "consumer provider".
From that point of perspective, this requirement should be removed.

Reply:
The notion of defining different roles for consumers and providers while both perspectives can be implemented in the same component makes sense to me. 
I would however, still state what each roles requires to do to make explicit what the roles mean. 

So for the specific case, I think it makes sense for a provider get the current values at least for the sensors that it provides since it might need to know the indicated value in the databroker.